src/windows/win-crowdstrike-fix-bootloop-v2.ps1

```diff
@@ -1,11 +1,124 @@
+            $driveLetter = ($driveLetter + ":")
+            Log-Info "Check Drive letter: $driveLetter"
+            $corruptFiles = "$driveLetter\Windows\System32\drivers\CrowdStrike\C-00000291*.sys"
+
+            if (Test-Path -Path $corruptFiles) {
+                Log-Info "Found crowdstrike files to cleanup at $corruptFiles, removing..."
+                Remove-Item $corruptFiles
+                Log-Info "Corrupt crowdstrike files are removed."
+                $crowdStrikeFileRemoved = $true
+            }
+        }
+    }
+
+    if ($crowdStrikeFileRemoved) {
+        Log-Info "Successfully cleaned up crowdstrike files"
+    } else {
+        Log-Warning "No bad crowdstrike files found"
+    }
+}
+
+# Check if registry config files exist in each non system drive letter.
+# if registry config files exist in the non system drive letter, load it to the registry hive and then unload it.
+function LoadUnloadRegistryHives
+{
+    param(
+    [Parameter(Mandatory = $true)]
+        [Object[]]$Partitionlist
+    )
+    Log-Info "Loading/unloading Registry Hives from registry config files..."
+
+    # System Drive (which is usually C:) should be skipped as it is from the OS disk rather than the Data disk
+    Log-Info "Getting system drive..."
+    $systemDrive = $Env:SYSTEMDRIVE
+    Log-Info "System drive is: $systemDrive"
+
+    $registryConfigFileFound = $false
+    forEach ( $partition in $partitionlist )
+    {
+        $driveLetter = $partition.DriveLetter
+        if ($driveLetter) { # Skip partitions without drive letter
+            $driveLetter = ($driveLetter + ":")
+            Log-Info "Check Drive letter: $driveLetter"
+            if ($driveLetter -ne $systemDrive) { # Skip OS disk
+                Log-Info "Found non system drive: $driveLetter"
+
+                Log-Info "Checking if registry config files exist from $driveLetter ..."
+                $configExixst = $false
+                $guidSuffix = "f85afa50-13cc-48e0-8a29-90603a43cfe1" # get a guid online as the reg key suffix in case the reg key name already exist
+                $regKeyToFile = @{
+                    "HKLM\temp_system_hive_$guidSuffix" = "$driveLetter\windows\system32\config\system"
+                    "HKLM\temp_software_hive_$guidSuffix" = "$driveLetter\windows\system32\config\software"
+                }
+
+                foreach ($regKey in $regKeyToFile.Keys)
+                {
+                    $regFile = $regKeyToFile[$regKey]
+                    if (Test-Path -Path $regFile) {
+                        Log-Info "Found registry config file at $regFile."
+                        $configExixst = $true
+
+                        Log-Info "Loading registry hive $regKey from $regFile..."
+                        $result = reg load $regKey $regFile 2>&1
+                        if ($LASTEXITCODE -ne 0) {
+                            Log-Error "Load registry hive $regKey from $regFile failed with error: $result"
```

Comment:
Should we log LASTEXITCODE as well?

Reply:
Sounds good. I just added $LASTEXITCODE into the log-Error